src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -790,6 +791,13 @@ public void Save (System.IO.TextWriter stream)
 			var s = new StreamReader (ms).ReadToEnd ();
 			if (ApplicationName != null)
 				s = s.Replace ("${applicationId}", ApplicationName);
+			if (Placeholders != null)
+				foreach (var entry in Placeholders.Split (';').Select (e => e.Split ('='))) {
```

Comment:
Splitting on `;` shouldn't be needed after moving to `string[]`. Additionally, the `e.Split('=')` should  instead use the `string.Split(char[], int, StringSplitOptions)` overload:

    static readonly char[] Separator = new [] { '=' };
    ...
    ... e.Split (Separator, 2, StringSplitOptions.RemoveEmptyEntries)